### PR TITLE
sqlbase,*: step 3 of removing sqlbase 

### PR DESCRIPTION
**Pull request by ajwerner**
First 3 are #53081. 

1 big commit that moves databases, system db, and keys out of sqlbase.

### CODE REVIEW

**Review by thoszhang — APPROVED** [review]

File: pkg/sql/catalog/catalogkeys/keys.go [hunk 1]
```diff
@@ -228,3 +154,201 @@ func PrettySpans(index *descpb.IndexDescriptor, spans []roachpb.Span, skip int)
...
+	parentSchemaID descpb.ID
+	name           string
+}
+
+// NewPublicTableKey returns a new TableKey scoped under the public schema.
+func NewPublicTableKey(parentID descpb.ID, name string) TableKey {
+	return TableKey{parentID: parentID, parentSchemaID: keys.PublicSchemaID, name: name}
+}
+
+// NewTableKey returns a new TableKey
```
Comment (lucy-zhang) [inline pkg/sql/catalog/catalogkeys/keys.go:190]: nit: these are missing some periods

File: pkg/sql/alter_schema.go [hunk 2]
```diff
@@ -26,7 +28,7 @@ import (
 
 type alterSchemaNode struct {
 	n    *tree.AlterSchema
-	db   *sqlbase.MutableDatabaseDescriptor
+	db   *dbdesc.MutableDatabaseDescriptor
```
Comment (rohany) [inline pkg/sql/alter_schema.go:31]: Is the goal to make this `dbdesc.MutableDescriptor`?

Comment (ajwerner) [reply pkg/sql/alter_schema.go:31]: have not decided, do you want it to be?

File: pkg/sql/catalog/catalogkeys/keys.go (hunk 1 above)
Comment (ajwerner) [reply pkg/sql/catalog/catalogkeys/keys.go:190]: added a commit to the end so I can just bors this one.

File: pkg/sql/alter_schema.go (hunk 2 above)
Comment (rohany) [reply pkg/sql/alter_schema.go:31]: maybe once we have packages for each kind of descriptor

File: pkg/sql/catalog/catalog.go [hunk 3]
```diff
@@ -100,3 +102,10 @@ type ResolvedSchema struct {
 	// SchemaUserDefined.
 	Desc SchemaDescriptor
 }
+
+// DescGetter is an interface to retrieve descriptors. In general the interface
```
Comment (ajwerner) [inline pkg/sql/catalog/catalog.go:106]: The MapDescGetter and general naming around that stuff moves into this package in a later commit. Please accept the partial measure in this commit.

File: pkg/sql/catalog/catalogkeys/keys.go (hunk 1 above)
Comment (ajwerner) [reply pkg/sql/catalog/catalogkeys/keys.go]: Done

**Review by rohany — APPROVED** [review]

**Pull request merged**